Documentation/boostrap-new-os.md

```diff
@@ -0,0 +1,140 @@
+  -crossgen               Crossgen the System.Private.CoreLib.dll, default: do not crossgen
+  -os <OS>                Operating system (used for corefx build), default: Linux
+  -rid <Rid>              Runtime identifier (without the architecture part)
+  -seedcli <SeedCli>      Seed CLI used to generate the target CLI
+```
+So, for example, when we were creating bootstrap CLI for RHEL / CentOS 6, the command was:
+```bash
+./buildbootstrapcli.sh -arch x64 -rid rhel.6 -release -os Linux -seedcli ~/seed-cli
+```
+After running the script, check the console output. If the last line printed is `**** Bootstrap CLI was successfully built  ****`, then everything went fine and the bootstrap CLI is ready. You can find it in the `<Rid>-<Architecture>/dotnetcli` subfolder. So for the example command above, it would be `rhel.6-x64/dotnetcli`.
+If there were build errors, they need to be looked into and fixed. After that run the `buildbootstrapcli.sh` with the same arguments again. Repeat until everything builds.
+### Testing the bootstrap CLI
+The easiest way to test the bootstrap CLI that was just created is to create, built and run a "hello world" console application. Create a new folder for the application and then run `/your/path/to/bootstrap/dotnetcli/dotnet new console` followed by `/your/path/to/bootstrap/dotnetcli/dotnet run`. If both of these succeed and the second prints "Hello world!", then everything went ok and you have a working bootstrap CLI. If there are issues in either of these steps, they need to be debugged and the culprits figured out.
+### Troubleshooting
+TODO: describe how to do debug build and replace System.Private.CoreLib.dll
+#### Using bootstrap CLI on platforms with different OS or architecture
+If the bootstrap CLI was created for OS other than the one the seed cli supports (e.g. FreeBSD) or the target architecture of the target OS differs from the architecture of the bootstrap cli (e.g. ARM64), the managed assemblies in the bootstrap CLI cannot be loaded, since they contain native code for the target OS and architecture. Fortunately, they also contain the original IL code, so they can be re-crossgened for the target OS and architecture. Or it is possible to instruct the CoreCLR runtime to ignore the native code and use the IL by setting environment variables `COMPlus_ZapDisable=1` and `COMPlus_ReadyToRun=0`. That can be useful during the bringup. But ultimately, re-crossgening should be made to improve startup performance.
+TODO: describe how to re-crossgen the managed assemblies
+## Building nuget packages
+Now that the bootstrap CLI works, it can be used to build coreclr, corefx and core-setup repos in its completeness, including the managed code. First set the `DOTNET_TOOL_DIR` env variable to the path where the boostrap CLI is located. That will cause the build process to use these tools instead of trying to download a `.tar.gz` package with OS specific tools from Azure.
```

Comment:
@weshaggard - I think we broke the `DOTNET_TOOL_DIR` scenario in the recent init-tools updates in `dev/release/2.0.0`.

Reply:
Yes see https://github.com/dotnet/corefx/blob/dev/release/2.0.0/init-tools.sh#L24 where we changed the property to be  `DotNetBuildToolsDir`.

Reply:
It's beyond that though.  In this case, we just want to take the .NET Core SDK that was already pre-built and use it.  Still download the BuildTools from wherever.

That's the scenario we broke.  You can't just use the .NET Core SDK from disk.  You have to have the whole BuildTools folder already populated.

Reply:
Fair. I filed https://github.com/dotnet/source-build/issues/229 to track that issue.

Reply:
Ah, I have thought the new env var was direct replacement for the DOTNET_TOOL_DIR and not for both that and the BUILD_TOOLS_TOOL_DIR.

Reply:
Looking closer as long as repo's respect __DOTNET_PATH then setting that before init-tools will get what we want here. https://github.com/dotnet/corefx/blob/dev/release/2.0.0/init-tools.sh#L40

Do you think that will be good enough or do we think we will need to have it copy/link local to the Tools directory similar to what we needed for buildtools?

Reply:
I don't think this will be good enough because the code inside our MSBuild scripts don't respect this env var.

https://github.com/dotnet/buildtools/blob/2523deb675aaea33ac93da7f03c369e3e49d919b/src/Microsoft.DotNet.Build.Tasks/PackageFiles/Build.Common.props#L38

BuildTools expects the folder to exist next to it in `dotnetcli`.